tests/codenames/test_codenames_game.py

```diff
@@ -0,0 +1,198 @@
+import unittest
+from playgroundrl_envs.games.codenames.codenames import (
+    CodenamesPlayer,
+    CodenamesGame,
+    CodenamesParameters,
+)
+
+from playgroundrl_envs.sid_util import SidSessionInfo
+import json, random
+
+
+def create_default_game():
+    params = CodenamesParameters()
+
+    session_info1 = SidSessionInfo(
+        sid="sid-0",
+        user_id=0,
+        is_human=False,
+    )
+
+    session_info2 = SidSessionInfo(
+        sid="sid-1",
+        user_id=1,
+        is_human=False,
+    )
+
+    players = [
+        CodenamesPlayer(
+            player_id=0,
+            session_info=session_info1,
+        ),
+        CodenamesPlayer(
+            player_id=1,
+            session_info=session_info2,
+        ),
+    ]
+
+    game = CodenamesGame(
+        game_id=0,
+        parameters=params,
+        players=players,
+        game_type=0,
+    )
+
+    return game
+
+def giver_valid_move(game):
+    state = json.loads(game.get_state()[0])  
+    return "valid", json.dumps({
+        "word": "xenophobic",  # random irrelevant word
+        "count": random.randint(1, 9),
+    })
+
+def giver_invalid_move(game):
+    state = json.loads(game.get_state()[0])  
+    error = random.randint(1, 3)
```

Comment:
Rather than do this randomly, we should do this as a for loop and test these conditions individually. When possible we should make deterministic tests with full coverage